libsplinter/src/oauth/store/memory.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Memory-backed InflightOAuthRequestStore implementation.
+
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use crate::oauth::{InflightOAuthRequestStore, PendingAuthorization};
+use crate::collections::TtlMap;
+use crate::error::InternalError;
+
+/// The amount of time before a pending authorization expires and a new request must be made
+const PENDING_AUTHORIZATION_EXPIRATION_SECS: u64 = 3600; // 1 hour
+
+#[derive(Clone)]
```

Comment:
Needs a doc string to provide some information about this implementation, especially around the timed expiration